### PR TITLE
add cwalk recipe

### PR DESCRIPTION
**Pull request by likle**
Specify library name and version:  **cwalk/1.2.2**, **cwalk/1.2.1**, **cwalk/1.2.0**, **cwalk/1.1.0**, **cwalk/1.0.0**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/cwalk/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename(self.name + "-" + self.version, self._source_subfolder)
+
+    def build(self):
+        cmake = CMake(self)
+        cmake.definitions["CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS"] = True
```
Comment (uilianries) [inline recipes/cwalk/all/conanfile.py]:
```suggestion
        cmake.definitions["CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS"] = self.options.shared and self.settings.os == "Windows"
```

File: recipes/cwalk/all/conandata.yml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+sources:
+  "1.2.2":
+    url: "https://github.com/likle/cwalk/archive/v1.2.2.zip"
+    sha256: "62094c2bb8882fd57fe7ea00fb196e6065b363e8ca57f9c693c19f4912747d88"
+  "1.2.1":
+    url: "https://github.com/likle/cwalk/archive/v1.2.1.zip"
+    sha256: "4bb7250e01a15f1c70e7f8d097e827d3f1a6ee77dcab3ee63de9c4bfd74fff67"
+  "1.2.0":
+    url: "https://github.com/likle/cwalk/archive/v1.2.0.zip"
+    sha256: "f74b4a97720ced5fac3e4507cab81274ae39d6844c16e319e81c02558c541805"
```
Comment (danimtb) [inline recipes/cwalk/all/conandata.yml]: Are all the 1.2.X needed? If not, please just leave the latest one

File: recipes/cwalk/all/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,7 @@
+cmake_minimum_required(VERSION 3.9.2)
```
Comment (danimtb) [inline recipes/cwalk/all/CMakeLists.txt]: If this the same version as the original cmakelists.txt of the library?

File: recipes/cwalk/config.yml [hunk 4]
```diff
@@ -0,0 +1,11 @@
+versions:
+  "1.2.2":
+    folder: all
+  "1.2.1":
+    folder: all
+  "1.2.0":
+    folder: all
```
Comment (danimtb) [inline recipes/cwalk/config.yml]: Same as above

File: recipes/cwalk/all/CMakeLists.txt (hunk 3 above)
Comment (likle) [reply recipes/cwalk/all/CMakeLists.txt]: yes it is! (https://github.com/likle/cwalk/blob/master/CMakeLists.txt)

**Pull request merged**